src/Utility/ChatTools.cs

```diff
@@ -0,0 +1,296 @@
+    /// <returns>A task representing the asynchronous operation.</returns>
+    public async Task AddMcpToolsAsync(McpClient client)
+    {
+        if (client == null) throw new ArgumentNullException(nameof(client));
+        _mcpClientsByEndpoint[client.Endpoint.AbsoluteUri] = client;
+        await client.StartAsync().ConfigureAwait(false);
+        BinaryData tools = await client.ListToolsAsync().ConfigureAwait(false);
+        await AddToolsAsync(tools, client).ConfigureAwait(false);
+        _mcpClients.Add(client);
+    }
+
+    /// <summary>
+    /// Adds a remote MCP server as a tool provider.
+    /// </summary>
+    /// <param name="mcpEndpoint">The URI endpoint of the MCP server.</param>
+    /// <returns>A task representing the asynchronous operation.</returns>
+    public async Task AddMcpToolsAsync(Uri mcpEndpoint)
+    {
+        var client = new McpClient(mcpEndpoint);
+        await AddMcpToolsAsync(client).ConfigureAwait(false);
+    }
+
+    private async Task AddToolsAsync(BinaryData toolDefinitions, McpClient client)
+    {
+        List<ChatTool> toolsToVectorize = new();
+        var parsedTools = ToolsUtility.ParseMcpToolDefinitions(toolDefinitions, client);
+
+        foreach (var (name, description, inputSchema) in parsedTools)
+        {
+            var chatTool = ChatTool.CreateFunctionTool(name, description, BinaryData.FromString(inputSchema));
+            _tools.Add(chatTool);
+            toolsToVectorize.Add(chatTool);
+            _mcpMethods[name] = client.CallToolAsync;
+        }
+
+        if (_client != null)
+        {
+            var embeddings = await _client.GenerateEmbeddingsAsync(toolsToVectorize.Select(t => t.FunctionDescription).ToList()).ConfigureAwait(false);
+            foreach (var embedding in embeddings.Value)
+            {
+                var vector = embedding.ToFloats();
+                var item = toolsToVectorize[embedding.Index];
+                var toolDefinition = SerializeTool(item);
+                _entries.Add(new VectorbaseEntry(vector, toolDefinition));
+            }
+        }
+    }
+
+    private BinaryData SerializeTool(ChatTool tool)
+    {
+        return ToolsUtility.SerializeTool(tool.FunctionName, tool.FunctionDescription, tool.FunctionParameters);
+    }
+
+    private ChatTool ParseToolDefinition(BinaryData data)
+    {
+        using var document = JsonDocument.Parse(data);
+        var root = document.RootElement;
+
+        return ChatTool.CreateFunctionTool(
+            root.GetProperty("name").GetString()!,
+            root.GetProperty("description").GetString()!,
+            BinaryData.FromString(root.GetProperty("inputSchema").GetRawText()));
+    }
+
+    /// <summary>
+    /// Converts the tools collection to chat completion options.
+    /// </summary>
+    /// <returns>A new ChatCompletionOptions containing all defined tools.</returns>
+    public ChatCompletionOptions CreateCompletionOptions()

```

Comment:
Since this method is not static, should we call ToChatCompletionOptions instead of CreateCompletionOptions?